statchatui/src/Compare/Compare.jsx

```diff
@@ -43,34 +37,45 @@ const Compare = ({ signedInUserData }) => {
     setIsLoading(true); //sets the state to true before the api call is displaying the stats
     let stats1;
     let stats2;
+    let validUser = true;
     try {
       stats1 = await fetchStats(user1);
       stats2 = await fetchStats(user2);
     } catch (error) {
       setError(error?.message);
+      validUser = false;
```

Comment:
If there's an error you should probably `return` here and not continue with the rest of the code. This way you don't need the `validUser` variable.